tomcat/src/main/java/org/apache/coyote/http11/Http11Processor.java

```diff
@@ -25,23 +35,121 @@ public void run() {
+            String requestLine = readRequestLine(inputStream);
+            if (requestLine == null || requestLine.isEmpty()) {
+                sendResponse(400, "Bad Request", "text/plain;charset=utf-8", new byte[0], outputStream);
+                return;
+            }
 
-            final var response = String.join("\r\n",
-                    "HTTP/1.1 200 OK ",
-                    "Content-Type: text/html;charset=utf-8 ",
-                    "Content-Length: " + responseBody.getBytes().length + " ",
-                    "",
-                    responseBody);
+            String[] request = requestLine.split(" ");
+            String fullPath = request[1];
+
+            String requestPath = extractPath(fullPath);
+            Map<String, String> queryParams = extractQueryParams(fullPath);
+
+            if ("/".equals(requestPath)) {
+                byte[] body = "Hello world!".getBytes(StandardCharsets.UTF_8);
+                sendResponse(200, "OK", "text/html;charset=utf-8", body, outputStream);
+            } else if ("/login".equals(requestPath)) {
+                handleLogin(queryParams, outputStream);
+            } else {
+                handleStaticResource(fullPath, requestPath, outputStream);
+            }
 
-            outputStream.write(response.getBytes());
-            outputStream.flush();
         } catch (IOException | UncheckedServletException e) {
             log.error(e.getMessage(), e);
         }
     }
+
+    private String readRequestLine(InputStream inputStream) throws IOException {
+        BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+        return reader.readLine();
+    }
+
+    private String extractPath(String fullPath) {
+        int index = fullPath.indexOf("?");
+
+        if (index != -1) {
+            return fullPath.substring(0, index);
+        }
+        return fullPath;
+    }
+
+    private Map<String, String> extractQueryParams(String fullPath) {
+        Map<String, String> queryParams = new HashMap<>();
+        int index = fullPath.indexOf("?");
+        if (index == -1) {
+            return queryParams;
+        }
+
+        String queryString = fullPath.substring(index + 1);
+        for (String pairs : queryString.split("&")) {
+            String[] pair = pairs.split("=", 2);
+            if (pair.length == 2) {
+                queryParams.put(pair[0], pair[1]);
+            }
+        }
+        return queryParams;
+    }
+
+    private void handleLogin(Map<String, String> queryParams, OutputStream outputStream) throws IOException {
+        String account = queryParams.get("account");
+        String password = queryParams.get("password");
+
+        InMemoryUserRepository.findByAccount(account).ifPresentOrElse(user -> {
+            if (user.checkPassword(password)) {
+                log.info("user: {}", user);
+            } else {
+                log.info("login fail(password mismatch) account: {}", account);
```

Comment:
로그인 실패시에는 실패 응답을 보내주면 좋을 것 같네요!
```suggestion
log.info("login fail(password mismatch) account: {}", account);
sendResponse(401, "Unauthorized", ~);
```

Reply:
좋은 것 같아요!

혹시 해당 부분 Login 구현하면서 반영해도 될까요? I/O Exception이 나면서 커스텀 예외를 만들어주고 싶은데 이후 요구사항에 로그인이 있어서요~!

Reply:
의도가 있으셨던 거군요~ 아직 3단계 남았으니 차근차근 반영해주세요!